vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -77,6 +77,27 @@ func patchVpaStatus(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName st
 	return vpaClient.Patch(context.TODO(), vpaName, types.ApplyPatchType, bytes, opts, "status")
 }
 
+func PatchVpaAnnotations(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string,
+	annotations map[string]string) error {
+
+	patches := make([]patchRecord, 0)
+	patches = append(patches, patchRecord{
+		Op:    "replace",
+		Path:  "/metadata/annotations",
+		Value: annotations,
+	})
+	bytes, err := json.Marshal(patches)
+	if err != nil {
+		return fmt.Errorf("Cannot marshal VPA annotations patches %+v. Reason: %+v", patches, err)
+	}
+	_, err = vpaClient.Patch(context.TODO(), vpaName, types.JSONPatchType, bytes, meta.PatchOptions{})
```

Comment:
We should try to apply the patch as server-side-apply, see: https://github.com/kubernetes/autoscaler/commit/a73c7520c6726792adc8b69d3a019112817357cf